generation/localizedstringkit/__init__.py

```diff
@@ -16,35 +16,44 @@
 from localizedstringkit.files import localizable_files
 
 
-SOURCE_CODE_FILE_NAME: str = "source_strings.m"
-BUNDLE_NAME: str = "LocalizedStringKit.bundle"
-
 log = logger.get()
 
 
-def generate_code_strings_file(code_files: List[str]) -> str:
-    """Generate a single code file with all strings.
+def generate_code_strings_file(code_files: List[str]) -> dict[str: str]:
+    """Generate a single code file with all strings per bundle.
 
     :param code_files: The list of file paths to generate the code strings for
 
-    :returns: The path to the temporary source code file with the standard NSLocalizedString calls
+    :returns: A dictionary of bundle name and path to the temporary source code file with the standard NSLocalizedString calls for that bundle
     """
 
-    output_path = tempfile.mktemp(suffix=".m")
-
+    # Get & dedupe localized strings
     localized_strings = detection.strings_in_code_files(code_files)
     localized_strings = list(set(localized_strings))
 
+    bundles: List[str] = []
+    for localized_string in localized_strings:
+        bundle = localized_string.bundle
+        if bundle is not None and bundle not in bundles:
+            bundles.append(bundle)
+
+    # Create output bundle and path dictionary for each unique bundle
+    output_paths: dict[str: str] = {}
+    for bundle in bundles:
+        output_paths[bundle] = tempfile.mktemp(suffix=".m")
+
     localized_strings.sort(key=lambda string: (string.key, string.key_extension, string.comment))
 
-    log.debug("Writing temporary source file at " + output_path)
+    for bundle, path in output_paths.items():
+        log.debug("Writing temporary source file at " + path + " for bundle " + bundle)
 
-    with open(output_path, "w") as temporary_source_file:
-        for localized_string in localized_strings:
-            temporary_source_file.write(localized_string.ns_localized_format())
-            temporary_source_file.write("\n")
+        with open(path, "w") as temporary_source_file:
+            for localized_string in localized_strings:
+                if localized_string.bundle == bundle:
```

Comment:
This works, but it's pretty inefficient. I'd put the strings into a dictionary based on their bundle name from the start. e.g.

```python
string_map = {}
for localized_string in localized_strings:
    if string_map.get(localized_string.bundle) is None:
        string_map[localized_string.bundle] = []
    string_map[localized_string.bundle].append(localized_string)

for bundle_name, localized_strings in string_map.items():
    with open(path_to_source_file) as whatever:
        for string in localized_strings:
            ...
```

Reply:
Yep, I planned on fixing these duplicate nested iteration. Updating